src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebRequestPSCmdlet.CoreClr.cs

```diff
@@ -408,6 +412,21 @@ protected override void StopProcessing()
 
         #region Helper Methods
 
+        private void ThrowNotSuccessStatusCodeError(HttpRequestMessage request, HttpResponseMessage response)
+        {
+            using(var reader = new StreamReader(StreamHelper.GetResponseStream(response)))
```

Comment:
This should handle any errors that can occur while obtaining and reading the response stream, such as invalid argument and IOException exceptions.